pkg/exporters/k8sexporter/problemclient/problem_client.go

```diff
@@ -125,6 +130,58 @@ func (c *nodeProblemClient) GetNode() (*v1.Node, error) {
 	return c.client.Nodes().Get(c.nodeName, metav1.GetOptions{})
 }
 
+// TaintNode taints the node if tainting is enabled and problem occurred
+func (c *nodeProblemClient) TaintNode(condition types2.Condition) error {
+	node, err := c.client.Nodes().Get(c.nodeName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	for _, v := range node.Spec.Taints {
+		if v.Key == condition.TaintKey && v.Value == condition.TaintValue && v.Effect == v1.TaintEffect(condition.TaintEffect) {
+			return nil
+		}
+	}
+
+	node.Spec.Taints = append(node.Spec.Taints, v1.Taint{
+		Key:       condition.TaintKey,
+		Value:     condition.TaintValue,
+		Effect:    v1.TaintEffect(condition.TaintEffect),
+	})
+
+	_, err = c.client.Nodes().Update(node)
```

Comment:
in my code, it will throw an error since resource conflict will occur. 

would not it be nice if we check the taint exists before trying the taint node? this way i guess no conflict will occur since second controller that trying to update taint will skip since taint already exists.

Reply:
each node-problem-detector instance will try to update its own taints using client. and each is not running async. why would there be a race condition in that scenario? sorry if i have missed some other point.